x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -226,23 +226,49 @@ export class JobCreator {
     this._calendars = calendars;
   }
 
+  private _initModelPlotConfig() {
+    // initialize configs to false if they are missing
+    if (this._job_config.model_plot_config === undefined) {
+      this._job_config.model_plot_config = {};
+    }
+    if (this._job_config.model_plot_config.enabled === undefined) {
+      this._job_config.model_plot_config.enabled = false;
+    }
+    if (this._job_config.model_plot_config.annotations_enabled === undefined) {
+      this._job_config.model_plot_config.annotations_enabled = false;
+    }
+  }
+
   public set modelPlot(enable: boolean) {
+    this._initModelPlotConfig();
     if (enable) {
-      this._job_config.model_plot_config = {
-        enabled: true,
-      };
+      this._job_config.model_plot_config.enabled = true;
```

Comment:
this can now be `this._job_config.model_plot_config.enabled = enable`

Reply:
Updated here https://github.com/elastic/kibana/pull/70678/commits/fcb1bab689c96967cc703f7eac92810988bfbd68